cmd/metal-api/internal/metal/switch.go

```diff
@@ -81,6 +92,13 @@ func SwitchModeFrom(name string) SwitchMode {
 	}
 }
 
+func ValidateSwitchOSVendor(os SwitchOSVendor) error {
+	if os != SwitchOSVendorCumulus && os != SwitchOSVendorSonic {
+		return fmt.Errorf("unknown switch os vendor %s", os)
+	}
+	return nil
```

Comment:
This is a bit easier to maintain because the linter figures out if it's exhaustive or not:

```suggestion
	switch os {
	case SwitchOSVendorCumulus, SwitchOSVendorSonic:
		return nil
	default:
		return fmt.Errorf("unknown switch os vendor %s", os)
	}
```